fccodegen/data/tests/weblab_model.pyx

```diff
@@ -193,3 +278,64 @@ cdef class GeneratedModel_Proto_tmpHzG5bQ(CvodeSolver):
         outputs[5][()] = var_environment__time
         return outputs
 
+    cpdef ResetState(self, name=None):
+        """
+        Reset the model to the given named saved state, or to initial
+        conditions if no name given.
+
+        See :meth:`fc.simulations.AbstractOdeModel.ResetState()`.
+        """
+        if name is None:
+            CvodeSolver.ResetSolver(self, self.initialState)
+        else:
+            CvodeSolver.ResetSolver(self, self.savedStates[name])
+
+    def SaveState(self, name):
+        """
+        Save a copy of the current model state associated with the given name,
+        to be restored using :meth:`ResetState()`.
+
+        See :meth:`fc.simulations.AbstractOdeModel.SaveState()`.
+        """
+        self.savedStates[name] = self.state.copy()
+
+    cpdef SetFreeVariable(self, double t):
+        """
+        Set the value of the free variable (typically time), but retain the
+        model's current state.
+
+        See :meth:`fc.simulations.AbstractOdeModel.SetFreeVariable()`.
+        """
+        self.freeVariable = t
+        CvodeSolver.SetFreeVariable(self, t)
+
+    def SetIndentLevel(self, indentLevel):
+        """
+        Set the level of indentation to use for progress output.
+
+        See :meth:`fc.simulations.AbstractModel.setIndentLevel()`.
+        """
+        self.indentLevel = indentLevel
+
+    def SetOutputFolder(self, path):
+        # TODO This is undocumented in fc
+        if os.path.isdir(path) and path.startswith('/tmp'):
+            shutil.rmtree(path)
+        os.mkdir(path)
+        self.outputPath = path
+
+    def SetRhsWrapper(self):
+        flag = Sundials.CVodeInit(
+            self.cvode_mem, _EvaluateRhs, 0.0, self._state)
+        self.CheckFlag(flag, "CVodeInit")
+
+    def SetSolver(self, solver):
+        """
+        Specify the ODE solver to use for this model.
+
+        See :meth:`fc.simulations.AbstractOdeModel.SetSolver()`.
+        """
+        # TODO Should this be python 3 syntax? Or is this special Cython
+        # syntax?
+        # TODO Can we use logging here instead? Or an exception?
```

Comment:
Those would be better options, yes. I can't remember why this doesn't raise; probably there's some test cases that trigger this path.